Add Create function to the dataset service

darrell-d opened the pull request:
- Add Create service to send create message to endpoint
- Add CreateDatasetResponse

darrell-d committed:
* Add Create service to send create message to endpoint
* Add CreateDatasetResponse

jwagenaar commented on pkg/pennsieve/dataset.go:
```diff
@@ -8,13 +8,15 @@ import (
 	Find(ctx context.Context, limit int, query string) (*dataset.ListDatasetResponse, error)
 	List(ctx context.Context, limit int, offset int) (*dataset.ListDatasetResponse, error)
 	SetBaseUrl(url string)
+	Create(apiKey, name, description string)
```
I think this is missing the ctx?

darrell-d replied:
Yes it was, also the function signature didn't match. Updated.

darrell-d committed:
Add correct funtion signature

darrell-d committed:
Add return type to function header

darrell-d committed:
Changes to the Create function

- Create the params from formatted string
- Allow https status code 201 (Created) to pass response error testing
